elements/rh-disclosure/rh-disclosure.ts

```diff
@@ -0,0 +1,67 @@
+import { LitElement, html } from 'lit';
+import { customElement } from 'lit/decorators/custom-element.js';
+import { property } from 'lit/decorators/property.js';
+import { query } from 'lit/decorators/query.js';
+
+import '@rhds/elements/rh-icon/rh-icon.js';
+
+import styles from './rh-disclosure.css';
+
+/**
+ * @summary A disclosure is a widget that enables content to be either collapsed (hidden) or expanded (visible).
+ * @slot - Place the content you want to disclose in the default slot. This content is hidden by default.
+ * @slot summary - The title of the disclosure
+ * @csspart caret - The caret icon in the shadow DOM
+ */
+@customElement('rh-disclosure')
+export class RhDisclosure extends LitElement {
+  static readonly styles = [styles];
+
+  /**
+   * Sets the disclosure to be in its open state
+   */
+  @property({ type: Boolean, reflect: true }) open = false;
+
+  @query('details') private detailsEl!: HTMLDetailsElement;
+  @query('summary') private summaryEl!: HTMLElement;
+
+  render() {
+    return html`
+      <details
+        ?open="${this.open}"
+        @keydown="${this.#onKeydown}"
+        @toggle="${this.#onToggle}">
+        <summary>
+          <rh-icon part="caret" id="caret" set="ui" icon="caret-down"></rh-icon>
+          <slot name="summary"></slot>
```

Comment:
```suggestion
          <slot name="summary">${this.summary}</slot>
```
we should make this an attr/slot pair like we did with [avatar](https://github.com/RedHat-UX/red-hat-design-system/blob/42aeb3c2762f9b1ac0feaff4e0f20f00e4f83344/elements/rh-avatar/rh-avatar.ts#L98), [switch](https://github.com/RedHat-UX/red-hat-design-system/blob/42aeb3c2762f9b1ac0feaff4e0f20f00e4f83344/elements/rh-switch/rh-switch.ts#L112), and [tooltip](https://github.com/RedHat-UX/red-hat-design-system/blob/42aeb3c2762f9b1ac0feaff4e0f20f00e4f83344/elements/rh-tooltip/rh-tooltip.ts#L82)

demos should prefer the attr, but we should have a dedicated demo for using the slot (i.e. slotting in an icon)